tests/main/validate-container-happy/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: check the symlinks following the right track
+
+environment:
+    SNAP: test-snapd-validate-container-happy
+
+prepare: |
+    snap pack test-snapd-validate-container-happy
+
+execute: |
+    snap try "$SNAP"
```

Comment:
hmm not sure about the order of operations here, I think that it may be better to instead have a few separate things to do and also have comments explaining what you are trying to test, I think the things that should be tested here are:

- [ ] that we can pack and install a snap that has all these weird symlinks in it (i.e. with the hell.tar extracted, then pack the snap and install the snap)
- [ ] that we can try a snap that has all these weird symlinks in it (i.e. with the hell.tar extracted, then `try` the snap)

In both cases, after the snap is installed/tried, check that the files in /snap/test-snapd-validate-container-happy/current/... have expected contents too